### ISSUE TITLE
Unit test fails due to SQL error when tests are run on mariaDB server

### ISSUE THREAD

**Issue by andrewmadden** [catalyst/moodle-tool_trigger#132]
There is a unit test failure when they are run using a `mariadb:10.5` docker container for the test DB.

```
There was 1 error:

1) tool_dataprivacy_expired_contexts_testcase::test_orphaned_records_are_cleared
Unexpected debugging() call detected.
Debugging: Error writing to database (You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'q
             WHERE NOT EXISTS (
                SELECT 1
                  ...' at line 2

            DELETE
              FROM phpu_tool_trigger_queue q
             WHERE NOT EXISTS (
                SELECT 1
                  FROM phpu_tool_trigger_events e
                 WHERE q.eventid = e.id
            )
[array (
)])
* line 1167 of /lib/dml/mysqli_native_moodle_database.php: call to moodle_database->query_end()
* line 210 of /admin/tool/trigger/classes/privacy/provider.php: call to mysqli_native_moodle_database->execute()
* line 187 of /admin/tool/trigger/classes/privacy/provider.php: call to tool_trigger\privacy\provider::delete_orphaned_queue_entries()
* line ? of unknownfile: call to tool_trigger\privacy\provider::delete_data_for_all_users_in_context()
* line 8107 of /lib/moodlelib.php: call to call_user_func_array()
* line 578 of /privacy/classes/manager.php: call to component_class_callback()
* line 611 of /privacy/classes/manager.php: call to core_privacy\manager::component_class_callback()
* line 492 of /privacy/classes/manager.php: call to core_privacy\manager->handled_component_class_callback()
* line 440 of /admin/tool/dataprivacy/classes/expired_contexts_manager.php: call to core_privacy\manager->delete_data_for_all_users_in_context()
* line 388 of /admin/tool/dataprivacy/classes/expired_contexts_manager.php: call to tool_dataprivacy\expired_contexts_manager->delete_expired_context()
* line 2070 of /admin/tool/dataprivacy/tests/expired_contexts_test.php: call to tool_dataprivacy\expired_contexts_manager->process_approved_deletions()
* line 1154 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to tool_dataprivacy_expired_contexts_testcase->test_orphaned_records_are_cleared()
* line 842 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestCase->runTest()
* line 80 of /lib/phpunit/classes/advanced_testcase.php: call to PHPUnit\Framework\TestCase->runBare()
* line 693 of /vendor/phpunit/phpunit/src/Framework/TestResult.php: call to advanced_testcase->runBare()
* line 796 of /vendor/phpunit/phpunit/src/Framework/TestCase.php: call to PHPUnit\Framework\TestResult->run()
* line 746 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestCase->run()
* line 746 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 746 of /vendor/phpunit/phpunit/src/Framework/TestSuite.php: call to PHPUnit\Framework\TestSuite->run()
* line 652 of /vendor/phpunit/phpunit/src/TextUI/TestRunner.php: call to PHPUnit\Framework\TestSuite->run()
* line 206 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\TestRunner->doRun()
* line 162 of /vendor/phpunit/phpunit/src/TextUI/Command.php: call to PHPUnit\TextUI\Command->run()
* line 61 of /vendor/phpunit/phpunit/phpunit: call to PHPUnit\TextUI\Command::main()


/var/www/site/lib/phpunit/classes/advanced_testcase.php:88

To re-run:
 vendor/bin/phpunit "tool_dataprivacy_expired_contexts_testcase" admin/tool/dataprivacy/tests/expired_contexts_test.php
```

**Comment by dmitriim** [comment]
I'd like to add extra privacy tests to CI for all of our plugins. It's not the first time I've seen something similar...

**Comment by andrewmadden** [comment]
I think the bug in this case, is that `derived tables` are not able to reference external contexts in mysql. In layman's terms, mysql doesn't know what 'q' is inside the subquery.
